test/parallel/xyk-pallet.sudo.security.test.ts

```diff
@@ -0,0 +1,190 @@
+    )
+
+    const result = await eventPromise;
+    expect(result.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+
+    const infoUser1After = await testUser1.getUserAccountInfo();
+    expect(infoUser1Before).not.toEqual(infoUser1After);
+    expect(infoUser1Before.free).toBeGreaterThan(infoUser1After.free);
+
+});
+
+test('xyk-pallet - SecurityTests - Only sudo can perform actions [tokens.create]', async () => {
+
+    const api = getApi();
+    let eventPromise = getUserEventResult("tokens","Issued", 12, testUser2.keyRingPair.address);
+
+    signTx(
+		api.tx.sudo.sudo(
+            api.tx.tokens.create(testUser2.keyRingPair.address, new BN(10000000))
+		),
+        testUser1.keyRingPair,
+        await getCurrentNonce(testUser1.keyRingPair.address)
+    );
+
+    const result = await eventPromise;
+    expect(result.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+
+});
+
+test('xyk-pallet - SecurityTests - Only sudo can perform actions [tokens.create to itself]', async () => {
+
+    const api = getApi();
+    let eventPromise = getUserEventResult("tokens","Issued", 12, testUser1.keyRingPair.address);
+
+    signTx(
+		api.tx.sudo.sudo(
+            api.tx.tokens.create(testUser1.keyRingPair.address, new BN(10000000))
+		),
+        testUser1.keyRingPair,
+        await getCurrentNonce(testUser1.keyRingPair.address)
+    );
+
+    const result = await eventPromise;
+    expect(result.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+
+});
+
+
+test('xyk-pallet - SecurityTests - Only sudo can perform actions [tokens.mint]', async () => {
+
+    const api = getApi();
+    const eventPromise = getUserEventResult("tokens", "Minted", 14, testUser2.keyRingPair.address);
+    signTx(
+        api.tx.sudo.sudo(
+          api.tx.tokens.mint(new BN(0), testUser2.keyRingPair.address, new BN(100000)),
+          ),
+        testUser1.keyRingPair,
+        await getCurrentNonce(testUser1.keyRingPair.address)   
+    )
+
+    const result = await eventPromise;
+    expect(result.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+
+});
+
+test('xyk-pallet - SecurityTests - Only sudo can perform actions [tokens.mint to itself]', async () => {
+
+    const api = getApi();
+    const eventPromise = getUserEventResult("tokens", "Minted", 14, testUser1.keyRingPair.address);
+    signTx(
+        api.tx.sudo.sudo(
+          api.tx.tokens.mint(new BN(0), testUser1.keyRingPair.address, new BN(100000)),
+          ),
+        testUser1.keyRingPair,
+        await getCurrentNonce(testUser1.keyRingPair.address)   
+    )
+
+    const result = await eventPromise;
+    expect(result.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+
+});
+
+afterEach(async () => {
+
+    const infoUser1After = await testUser1.getUserAccountInfo();
+    expect(infoUser1Before).not.toEqual(infoUser1After);
+    expect(infoUser1Before.free).toBeGreaterThan(infoUser1After.free);
+    await waitNewBlock();
+    await waitNewBlock();
```

Comment:
Why is the before different from after if the extrinsic failed?
What am I missing here?
Is it because of txn fee or nonce?

Reply:
The fee Is consumed even if Tx fail. 
Hope is the right behavior :)